src/middlewares/authorization/AuthorizationFactory.ts

```diff
@@ -0,0 +1,65 @@
+import { singleton, inject } from 'tsyringe';
+
+import { AppUserRole } from '@Entities';
+import { AuthenticationService, AuthorizationService } from '@Services';
+import { AuthorizationMiddleware } from './AuthorizationMiddleware';
+import * as ERR_MSGS from '../../constants/ErrResponses';
+
+@singleton()
+export class AuthorizationFactory {
+  private authenticationService: AuthenticationService;
+  private authorizationService: AuthorizationService;
+
+  constructor(
+    @inject(AuthenticationService) authenticationService: AuthenticationService,
+    @inject(AuthorizationService) authorizationService: AuthorizationService
+  ) {
+    this.authenticationService = authenticationService;
+    this.authorizationService = authorizationService;
+  }
+
+  getAuthorizationMiddleware(permissionLevel: string): AuthorizationMiddleware {
+    const { authenticationService, authorizationService } = this;
```

Comment:
jw, does eslint complain if you don't do destructuring?

Reply:
I'm not sure, I just like destructuring